server/game/core/gameSystem/GameSystem.ts

```diff
@@ -196,11 +199,26 @@ export abstract class GameSystem<TContext extends AbilityContext = AbilityContex
     public queueGenerateEventGameSteps(events: GameEvent[], context: TContext, additionalProperties: any = {}): void {
         for (const target of this.targets(context, additionalProperties)) {
             if (this.canAffect(target, context, additionalProperties)) {
-                events.push(this.generateEvent(target, context, additionalProperties));
+                events.push(this.generateRetargetedEvent(target, context, additionalProperties));
             }
         }
     }
 
+    /**
+     * Generates one {@link GameEvent} object that will apply the effects of this system to the game state
+     * for the specified target.
+     * The event must be emitted using an {@link EventWindow}, typically via `Game.openEventWindow`.
+     * @param context Context of ability being executed
+     * @param additionalProperties Any additional properties to extend the default ones with
+     */
+    public generateEvent(context: TContext, additionalProperties: any = {}): GameEvent {
```

Comment:
Instead of using just `any` here, we can probably use a more specific type like `Record<string, any>` since we expect a hash.